### ISSUE TITLE
[Bug]: Building lakeFS on MacOS results in many 0byte files added

### ISSUE THREAD

**Issue by ozkatz** [treeverse/lakeFS#7694]
### What happened?

What actually happened, including error codes if applies.
        
Steps to Reproduce:
1. clone lakeFS
2. run `make build`

### Expected behavior

lakeFS binaries appear, no untracked files added to `git status`

### lakeFS version

master

### How lakeFS is installed

locally

### Relevant log output

```shell
$ git status  # after `make build` successfully finishes
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   clients/python/lakefs_sdk/api/actions_api.py
	modified:   clients/python/lakefs_sdk/api/auth_api.py
	modified:   clients/python/lakefs_sdk/api/branches_api.py
	modified:   clients/python/lakefs_sdk/api/commits_api.py
	modified:   clients/python/lakefs_sdk/api/config_api.py
	modified:   clients/python/lakefs_sdk/api/experimental_api.py
	modified:   clients/python/lakefs_sdk/api/external_api.py
	modified:   clients/python/lakefs_sdk/api/health_check_api.py
	modified:   clients/python/lakefs_sdk/api/import_api.py
	modified:   clients/python/lakefs_sdk/api/internal_api.py
	modified:   clients/python/lakefs_sdk/api/metadata_api.py
	modified:   clients/python/lakefs_sdk/api/objects_api.py
	modified:   clients/python/lakefs_sdk/api/refs_api.py
	modified:   clients/python/lakefs_sdk/api/repositories_api.py
	modified:   clients/python/lakefs_sdk/api/staging_api.py
	modified:   clients/python/lakefs_sdk/api/tags_api.py
	modified:   clients/python/lakefs_sdk/api_response.py
	modified:   clients/python/lakefs_sdk/models/abort_presign_multipart_upload.py
	modified:   clients/python/lakefs_sdk/models/access_key_credentials.py
	modified:   clients/python/lakefs_sdk/models/acl.py
	modified:   clients/python/lakefs_sdk/models/action_run.py
	modified:   clients/python/lakefs_sdk/models/action_run_list.py
	modified:   clients/python/lakefs_sdk/models/auth_capabilities.py
	modified:   clients/python/lakefs_sdk/models/authentication_token.py
	modified:   clients/python/lakefs_sdk/models/branch_creation.py
	modified:   clients/python/lakefs_sdk/models/branch_protection_rule.py
	modified:   clients/python/lakefs_sdk/models/cherry_pick_creation.py
	modified:   clients/python/lakefs_sdk/models/comm_prefs_input.py
	modified:   clients/python/lakefs_sdk/models/commit.py
	modified:   clients/python/lakefs_sdk/models/commit_creation.py
	modified:   clients/python/lakefs_sdk/models/commit_list.py
	modified:   clients/python/lakefs_sdk/models/commit_record_creation.py
	modified:   clients/python/lakefs_sdk/models/complete_presign_multipart_upload.py
	modified:   clients/python/lakefs_sdk/models/config.py
	modified:   clients/python/lakefs_sdk/models/credentials.py
	modified:   clients/python/lakefs_sdk/models/credentials_list.py
	modified:   clients/python/lakefs_sdk/models/credentials_with_secret.py
	modified:   clients/python/lakefs_sdk/models/current_user.py
	modified:   clients/python/lakefs_sdk/models/diff.py
	modified:   clients/python/lakefs_sdk/models/diff_list.py
	modified:   clients/python/lakefs_sdk/models/error.py
	modified:   clients/python/lakefs_sdk/models/error_no_acl.py
	modified:   clients/python/lakefs_sdk/models/external_login_information.py
	modified:   clients/python/lakefs_sdk/models/external_principal.py
	modified:   clients/python/lakefs_sdk/models/external_principal_creation.py
	modified:   clients/python/lakefs_sdk/models/external_principal_list.py
	modified:   clients/python/lakefs_sdk/models/find_merge_base_result.py
	modified:   clients/python/lakefs_sdk/models/garbage_collection_config.py
	modified:   clients/python/lakefs_sdk/models/garbage_collection_prepare_response.py
	modified:   clients/python/lakefs_sdk/models/garbage_collection_rule.py
	modified:   clients/python/lakefs_sdk/models/garbage_collection_rules.py
	modified:   clients/python/lakefs_sdk/models/group.py
	modified:   clients/python/lakefs_sdk/models/group_creation.py
	modified:   clients/python/lakefs_sdk/models/group_list.py
	modified:   clients/python/lakefs_sdk/models/hook_run.py
	modified:   clients/python/lakefs_sdk/models/hook_run_list.py
	modified:   clients/python/lakefs_sdk/models/import_creation.py
	modified:   clients/python/lakefs_sdk/models/import_creation_response.py
	modified:   clients/python/lakefs_sdk/models/import_location.py
	modified:   clients/python/lakefs_sdk/models/import_status.py
	modified:   clients/python/lakefs_sdk/models/installation_usage_report.py
	modified:   clients/python/lakefs_sdk/models/internal_delete_branch_protection_rule_request.py
	modified:   clients/python/lakefs_sdk/models/login_config.py
	modified:   clients/python/lakefs_sdk/models/login_information.py
	modified:   clients/python/lakefs_sdk/models/merge.py
	modified:   clients/python/lakefs_sdk/models/merge_result.py
	modified:   clients/python/lakefs_sdk/models/meta_range_creation.py
	modified:   clients/python/lakefs_sdk/models/meta_range_creation_response.py
	modified:   clients/python/lakefs_sdk/models/object_copy_creation.py
	modified:   clients/python/lakefs_sdk/models/object_error.py
	modified:   clients/python/lakefs_sdk/models/object_error_list.py
	modified:   clients/python/lakefs_sdk/models/object_stage_creation.py
	modified:   clients/python/lakefs_sdk/models/object_stats.py
	modified:   clients/python/lakefs_sdk/models/object_stats_list.py
	modified:   clients/python/lakefs_sdk/models/pagination.py
	modified:   clients/python/lakefs_sdk/models/path_list.py
	modified:   clients/python/lakefs_sdk/models/policy.py
	modified:   clients/python/lakefs_sdk/models/policy_list.py
	modified:   clients/python/lakefs_sdk/models/prepare_gc_uncommitted_request.py
	modified:   clients/python/lakefs_sdk/models/prepare_gc_uncommitted_response.py
	modified:   clients/python/lakefs_sdk/models/presign_multipart_upload.py
	modified:   clients/python/lakefs_sdk/models/range_metadata.py
	modified:   clients/python/lakefs_sdk/models/ref.py
	modified:   clients/python/lakefs_sdk/models/ref_list.py
	modified:   clients/python/lakefs_sdk/models/refs_dump.py
	modified:   clients/python/lakefs_sdk/models/refs_restore.py
	modified:   clients/python/lakefs_sdk/models/repository.py
	modified:   clients/python/lakefs_sdk/models/repository_creation.py
	modified:   clients/python/lakefs_sdk/models/repository_dump_status.py
	modified:   clients/python/lakefs_sdk/models/repository_list.py
	modified:   clients/python/lakefs_sdk/models/repository_metadata_keys.py
	modified:   clients/python/lakefs_sdk/models/repository_metadata_set.py
	modified:   clients/python/lakefs_sdk/models/repository_restore_status.py
	modified:   clients/python/lakefs_sdk/models/reset_creation.py
	modified:   clients/python/lakefs_sdk/models/revert_creation.py
	modified:   clients/python/lakefs_sdk/models/setup.py
	modified:   clients/python/lakefs_sdk/models/setup_state.py
	modified:   clients/python/lakefs_sdk/models/staging_location.py
	modified:   clients/python/lakefs_sdk/models/staging_metadata.py
	modified:   clients/python/lakefs_sdk/models/statement.py
	modified:   clients/python/lakefs_sdk/models/stats_event.py
	modified:   clients/python/lakefs_sdk/models/stats_events_list.py
	modified:   clients/python/lakefs_sdk/models/storage_config.py
	modified:   clients/python/lakefs_sdk/models/storage_uri.py
	modified:   clients/python/lakefs_sdk/models/sts_auth_request.py
	modified:   clients/python/lakefs_sdk/models/tag_creation.py
	modified:   clients/python/lakefs_sdk/models/task_info.py
	modified:   clients/python/lakefs_sdk/models/underlying_object_properties.py
	modified:   clients/python/lakefs_sdk/models/update_token.py
	modified:   clients/python/lakefs_sdk/models/upload_part.py
	modified:   clients/python/lakefs_sdk/models/usage_report.py
	modified:   clients/python/lakefs_sdk/models/user.py
	modified:   clients/python/lakefs_sdk/models/user_creation.py
	modified:   clients/python/lakefs_sdk/models/user_list.py
	modified:   clients/python/lakefs_sdk/models/version_config.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	clients/python/lakefs_sdk/api/sed08JOU3
	clients/python/lakefs_sdk/api/sed30t9yr
	clients/python/lakefs_sdk/api/sed4xc4fF
	clients/python/lakefs_sdk/api/sed51pSoN
	clients/python/lakefs_sdk/api/sedAHG0dQ
	clients/python/lakefs_sdk/api/sedBvVKNE
	clients/python/lakefs_sdk/api/sedCqLARs
	clients/python/lakefs_sdk/api/sedEepFTD
	clients/python/lakefs_sdk/api/sedIxbFZq
	clients/python/lakefs_sdk/api/sedVSZ5Ic
	clients/python/lakefs_sdk/api/sedeTpo95
	clients/python/lakefs_sdk/api/sedgHF48O
	clients/python/lakefs_sdk/api/sedidksBZ
	clients/python/lakefs_sdk/api/seds1M9Mb
	clients/python/lakefs_sdk/api/sedtqn0YD
	clients/python/lakefs_sdk/api/sedufKiEf
	clients/python/lakefs_sdk/models/sed2PKNNA
	clients/python/lakefs_sdk/models/sed31AK1b
	clients/python/lakefs_sdk/models/sed6FBuoz
	clients/python/lakefs_sdk/models/sed6T11Bd
	clients/python/lakefs_sdk/models/sed6ZP2RX
	clients/python/lakefs_sdk/models/sed76aHyg
	clients/python/lakefs_sdk/models/sed7OYDmT
	clients/python/lakefs_sdk/models/sed8CNM7Y
	clients/python/lakefs_sdk/models/sed9PPZEl
	clients/python/lakefs_sdk/models/sedAbCbTD
	clients/python/lakefs_sdk/models/sedBsjGuX
	clients/python/lakefs_sdk/models/sedC27qws
	clients/python/lakefs_sdk/models/sedClYHhq
	clients/python/lakefs_sdk/models/sedD8VegO
	clients/python/lakefs_sdk/models/sedDmM6D6
	clients/python/lakefs_sdk/models/sedEJh40p
	clients/python/lakefs_sdk/models/sedExiBTB
	clients/python/lakefs_sdk/models/sedFiDm5X
	clients/python/lakefs_sdk/models/sedGVYF7N
	clients/python/lakefs_sdk/models/sedGlC5Xx
	clients/python/lakefs_sdk/models/sedH8iI4x
	clients/python/lakefs_sdk/models/sedHr3R2A
	clients/python/lakefs_sdk/models/sedJAIK3v
	clients/python/lakefs_sdk/models/sedJmubcL
	clients/python/lakefs_sdk/models/sedK6OHNC
	clients/python/lakefs_sdk/models/sedK7B8et
	clients/python/lakefs_sdk/models/sedK9bgAm
	clients/python/lakefs_sdk/models/sedKZaTNM
	clients/python/lakefs_sdk/models/sedKvZMft
	clients/python/lakefs_sdk/models/sedLdO5Kh
	clients/python/lakefs_sdk/models/sedLitFUp
	clients/python/lakefs_sdk/models/sedM0hSpS
	clients/python/lakefs_sdk/models/sedOBVrPK
	clients/python/lakefs_sdk/models/sedOYuADE
	clients/python/lakefs_sdk/models/sedPOx5Pa
	clients/python/lakefs_sdk/models/sedQ0SKF7
	clients/python/lakefs_sdk/models/sedQ9unwb
	clients/python/lakefs_sdk/models/sedQDlKc1
	clients/python/lakefs_sdk/models/sedQN9OcD
	clients/python/lakefs_sdk/models/sedQQ63N2
	clients/python/lakefs_sdk/models/sedRYfKUc
	clients/python/lakefs_sdk/models/sedS7ct9h
	clients/python/lakefs_sdk/models/sedSfWe01
	clients/python/lakefs_sdk/models/sedTCGcUZ
	clients/python/lakefs_sdk/models/sedU1hz04
	clients/python/lakefs_sdk/models/sedUL8cy0
	clients/python/lakefs_sdk/models/sedVPK7Sd
	clients/python/lakefs_sdk/models/sedWCm7WL
	clients/python/lakefs_sdk/models/sedWFWK7L
	clients/python/lakefs_sdk/models/sedYOWO9V
	clients/python/lakefs_sdk/models/sedYcHKC3
	clients/python/lakefs_sdk/models/sedZY2wFO
	clients/python/lakefs_sdk/models/sedZzrStc
	clients/python/lakefs_sdk/models/sedaQ57hn
	clients/python/lakefs_sdk/models/sedaq0BFP
	clients/python/lakefs_sdk/models/sedaz8E9w
	clients/python/lakefs_sdk/models/sedbRFLxZ
	clients/python/lakefs_sdk/models/sedbiOE5l
	clients/python/lakefs_sdk/models/sedbqpwIG
	clients/python/lakefs_sdk/models/sedchndRq
	clients/python/lakefs_sdk/models/seddELQAQ
	clients/python/lakefs_sdk/models/seddIEw2q
	clients/python/lakefs_sdk/models/sedeD87IE
	clients/python/lakefs_sdk/models/sedf6LKlH
	clients/python/lakefs_sdk/models/sedgkBzue
	clients/python/lakefs_sdk/models/sedh0YgkW
	clients/python/lakefs_sdk/models/sediUbJgU
	clients/python/lakefs_sdk/models/sedjgsaUk
	clients/python/lakefs_sdk/models/sedkZ68F6
	clients/python/lakefs_sdk/models/sedkhxZHN
	clients/python/lakefs_sdk/models/sedn4IwcV
	clients/python/lakefs_sdk/models/sedorG70t
	clients/python/lakefs_sdk/models/sedpPmlNg
	clients/python/lakefs_sdk/models/sedpbxWaJ
	clients/python/lakefs_sdk/models/sedpnbuDn
	clients/python/lakefs_sdk/models/sedq0ljta
	clients/python/lakefs_sdk/models/sedqmW1NF
	clients/python/lakefs_sdk/models/sedrQxYU4
	clients/python/lakefs_sdk/models/sedrvU4DE
	clients/python/lakefs_sdk/models/seds3MUWn
	clients/python/lakefs_sdk/models/sedsVzAEr
	clients/python/lakefs_sdk/models/sedsbDHBN
	clients/python/lakefs_sdk/models/sedtZVqsa
	clients/python/lakefs_sdk/models/sedtcQ4YR
	clients/python/lakefs_sdk/models/sedtj2m7h
	clients/python/lakefs_sdk/models/sedtq7gQO
	clients/python/lakefs_sdk/models/sedu5PIJ4
	clients/python/lakefs_sdk/models/seduEBnVA
	clients/python/lakefs_sdk/models/seduFvD2D
	clients/python/lakefs_sdk/models/sedvEuFyj
	clients/python/lakefs_sdk/models/sedvhkzNN
	clients/python/lakefs_sdk/models/sedw2AHSS
	clients/python/lakefs_sdk/models/sedw944A8
	clients/python/lakefs_sdk/models/sedxbSqRw
	clients/python/lakefs_sdk/models/sedxepZl1
	clients/python/lakefs_sdk/models/sedxlr0hf
	clients/python/lakefs_sdk/models/sedyDnKAg
	clients/python/lakefs_sdk/models/sedyQeTF6
	clients/python/lakefs_sdk/sedYF0jDu

no changes added to commit (use "git add" and/or "git commit -a")
```

**Comment by N-o-Z** [comment]
@ozkatz Beside the additionally created files, does the sed command work? Are the python import statements modified appropriately?

**Comment by ozkatz** [comment]
I'm closing this for now, as I don't believe it will affect other people going forward. This was a pretty deep rabbit hole.
*TL;DR - On a Mac, make sure your Docker Desktop is up to date.*

Longer version: Docker Desktop on Mac switched to mounting local directories with [VirtioFS](https://virtio-fs.gitlab.io/) a while back. Up until recent versions, that came with a small caveat - mounted volumes failed to properly create a file with mode=0.

This should be a pretty obscure thing, since why would you want a file that isn't readable, writable or executable by anyone?

Well, turns out, [GNU sed](https://www.gnu.org/software/sed/) has a bug - when doing `sed -i` on a file, it would create a randomly named file first, apply the replacements there, and then atomically overwrite the original file. So far so good - however, it would first `umask(0700) and then open the temporary file with mode `0600`, resulting in a mode of 0.

The combination of the 2 issues led to a bunch of sed temporary files, and python client that is broken for me.

I upgraded my Docker Desktop version and now it works :(

Thanks @N-o-Z @guy-har for the attention.
